src/AppInstallerCommonCore/Public/winget/ManifestCommon.h

```diff
@@ -223,7 +224,15 @@ namespace AppInstaller::Manifest
             return MinVersion <= Utility::Version(version);
         }
 
+        // m_foldedId should be set whenever Id is set

```

Comment:
So, the issue was that in some cases it wasn't set?

Reply:
Correct. Before the fix, manifest parsing directly sets Id without anyway to set m_foldedId resulting in all dependency packages having same folded id and get treated as same package.